src/test/java/com/google/devtools/build/lib/analysis/platform/ConstraintCollectionTest.java

```diff
@@ -0,0 +1,82 @@
+package com.google.devtools.build.lib.analysis.platform;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.cmdline.Label;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests of {@link ConstraintCollection}. */
+@RunWith(JUnit4.class)
+public class ConstraintCollectionTest extends BuildViewTestCase {
+  @Test
+  public void testFindMissing() {
```

Comment:
This doesn't really just test findMissing but also containsAll. Perhaps rename to testSetArithmetic? This also makes me think that the current naming could possibly be improved by using math terms like "difference" rather than findMissing (although the latter is also fairly obvious).